daemon/compartment.c

```diff
@@ -1209,6 +1217,16 @@ compartment_start_post_clone_cb(int fd, unsigned events, event_io_t *io, void *d
 		WARN_ERRNO("write to sync socket failed");
 		goto error;
 	}
+
+	// kill running helper children
+	for (list_t *l = compartment->helper_child_list; l; l = l->next) {
+		compartment_helper_child_t *child = l->data;
+		if (kill(child->pid, SIGKILL)) {
```

Comment:
we should ensure pid > 0 here

Reply:
should be the conservative approach yes.